### ISSUE TITLE
umb-overlay inside umb-editor is hidden

### ISSUE THREAD

**Issue by enkelmedia** [umbraco/Umbraco-CMS#7754]
We're using some "Nested Nested Content" inside Doc Type Grid Editor, this works fine until there is more than one Document Type configured on the Nested Content Data Type. In these cases a "umb-overlay" is shown so that the editor can pick which DocType to add, like this:

![screen - nested content choose type](https://user-images.githubusercontent.com/1782524/75871313-a5a69e00-5e0c-11ea-80c1-e31a0fe69828.PNG)

This works all fine, but when Nested Content is shown inside a EditorService-dialog, the umb-overlay will set it's style to left:xxx px; so that it ends up outside the screen, this is how it looks when clicking "Add Content"

![screen - nested in editor service](https://user-images.githubusercontent.com/1782524/75871616-21a0e600-5e0d-11ea-9c61-1a04de8da3c9.PNG)

After removing the "left"-style from the umb-overlay the box shows again:

![screen nested inside editorservice removed left](https://user-images.githubusercontent.com/1782524/75871666-34b3b600-5e0d-11ea-8de7-1d6688aa1fec.PNG)

This issue is present only with "Doc Type Grid Editor" but also when for example configuring a "Multi Node Tree Picker" to have the "Open"-button. When opening an Node and trying to choose a DocType for Nested Content it looks like this:

![screen - omb-overlay in editor service edit content](https://user-images.githubusercontent.com/1782524/75872082-ddfaac00-5e0d-11ea-8765-e6e7b36dd77f.PNG)

In other words, the umb-overlay is placed to much to the right so that the editor can't choose a DocType. The issue is specially big when the editorService is opening a window that is of type "small".

We've solved this with some CSS-hacks but I guess it would be good to come up with something in Core?

Our solution was to put any umb-overlays to the right using CSS like this but there might be better solutions =D

```css
.umb-editors {
    .umb-overlay {
        left: inherit !important;
        right: 0 !important;
        top: 0 !important;
        bottom: 0 !important;
    }
}
```

**Comment by nul800sebastiaan** [comment]
Just make sure to always mention the Umbraco version you're on please so we can try to reproduce!

**Comment by abjerner** [comment]
We have experienced this as well, but I haven't gotten around to creating a PR or even creating an issue yet.

I think this is in 8.5, but not in 8.4. But I haven't really tested beyond this.

Umbraco currently uses the `<umb-overlay/>` directive, which is deprecated. It has some quirks - eg. it isn't positioned correctly when used in infinite editing. I think the the fix should be updating the code to use the newer `overlayService` or `editorService` instead.

**Comment by nul800sebastiaan** [comment]
Good news, I just tested in v8.6 and it's already fixed there: 

![image](https://user-images.githubusercontent.com/304656/75896142-e74c3e80-5e36-11ea-9907-f51fd0c23570.png)

**Comment by abjerner** [comment]
@nul800sebastiaan your editor is full size. As can be seen on your screenshot, the overlay isn't fully grid centered, which I guess is also part of the issue.

When using a `medium` or `small` editor, the overlay still seems to be off screen. I just tested this with the latest `v8/contrib`.

![umbOverlay](https://user-images.githubusercontent.com/3634580/75901116-58432480-5e3e-11ea-809d-a76c68602fed.gif)

If we switch to using the `overlayService`, I think it will always be horizontally centered based on the overall viewport, whereas I think the idea with the current `umbOverlay` directive is that it should be positioned relative to the active infinite editor (although it doesn't work). Would that be an acceptable trade off if switching to the `overlayService`?

**Comment by abjerner** [comment]
Here is an example on how it will look using the similar `itemPicker` in the `overlayService` (as mentioned, the overlay will be positioned in relation to the overall viewport, and not the active editor):

![overlayService](https://user-images.githubusercontent.com/3634580/75903023-3dbe7a80-5e41-11ea-85ed-835d6977f143.gif)

**Comment by enkelmedia** [comment]
@nul800sebastiaan Sorry, this was on Umbraco 8.4 and @abjerner's screenshare shows the issue that I'm reffering to when the editorService is opened in a "small" window.

**Comment by nul800sebastiaan** [comment]
Thanks @abjerner - Apparently I didn't know how to reproduce correctly! 👍

**Comment by nul800sebastiaan** [comment]
Fixed in https://github.com/umbraco/Umbraco-CMS/pull/7759

**Comment by stefankip** [comment]
I think https://github.com/umbraco/Umbraco-CMS/issues/7918 is the same issue and thus duplicate, right?

**Comment by abjerner** [comment]
@stefankip Yes and no.

It's the same underlying problem, but my PR only fixes this for the grid overlay only, as it doesn't fix the problem with the `umb-overlay` directive, but instead replaces the use of the directive with the newer `overlayService`.

I can't really remember for sure whether the problem was also fixed generally for the `umb-overlay` directive. If not, it would be useful as it would fix the problem globally for the directive. And it may be a lot quicker than replacing all uses of `umb-overlay` with `overlayService`.

**Comment by enkelmedia** [comment]
I've shared a "fix" that works for us here https://github.com/umbraco/Umbraco-CMS/issues/7918#issuecomment-625982546 and maybe it could be a starting point for something to ship in core.
